src/server/services/wfs/qgswfsutils.h

```diff
@@ -60,7 +60,7 @@ namespace QgsWfs
   /**
    * Transform a Filter element to a feature request
    */
-  QgsFeatureRequest parseFilterElement( const QString &typeName, QDomElement &filterElem, const QgsProject *project = nullptr );
+  QgsFeatureRequest parseFilterElement( const QString &typeName, QDomElement &filterElem, const QgsProject *project = nullptr, QStringList fids = QStringList() );
```

Comment:
I think that `fids` should be a const reference in this case.

Reply:
It can't be a const, because it will be updated by the method if featureIds are found in the filter element.

Reply:
But passing arguments by value is not a good practice no?

Reply:
What is normally done in cases like this is passing it as const ref and then taking a local copy.

Reply:
> What is normally done in cases like this is passing it as const ref and then taking a local copy.

+1

Reply:
Why is the feature id list a string list? I think it should either be `QgsFeatureIds` or renamed to something else than feature id (primary key?). Or I might be totally missing a point because I didn't review in depth at all.

Reply:
We can't use QgsFeatureIds in QGIS Server because in WFS and in WMS GetFeatureInfo (format GML), there is a 'fid' Attribute with a feature identification. Currently, we are using the QGIS feature id there. However, the postgres provider uses an id/value map for some data types (e.g. bigint in 2.18). Therefore, the feature id depends on the fetching order and might not be stable between different requests. So clients cannot use the fid for feature identification.

Reply:
If I use `fids` as `const`, I can't modifiy it, is it ? @pblottiere @m-kuhn

Reply:
* I have read the pull request description too, as far as I can see it you are mostly repeating the reason for this pull request in all (and stable ids are desirable here) however, I am missing the point how this is related to my question.
* Yes, const cannot be modified. But a non-const copy of a const object can.

Reply:
This method extracted the fid used in filter element. So the fids are not simple integer but strings as they have been provided by QGIS Server. We can't directly used them in FeatureRequest without the layer context.

Reply:
In this case would the second part of my [above proposal](https://github.com/qgis/QGIS/pull/10034#discussion_r286357054) make sense? I.e. is it worth renaming the variable to something like primaryKeys so we keep the term `fid` strictly for internal ids of type QgsFeatureId?

Reply:
serverFid because of QgsServerFeatureid ?

Reply:
Yes, sounds better!

Reply:
Updated to use const referecne for serverFids.